homeassistant/scripts/check_config.py

```diff
@@ -0,0 +1,134 @@
+"""Script to ensure a configuration file exists."""
+import argparse
+import os
+import glob
+# from typing import Optional, Dict
+from unittest.mock import patch
+import importlib
+
+import homeassistant.bootstrap as bootstrap
+import homeassistant.config as config_util
+import homeassistant.util.yaml as yaml_util
+import homeassistant.loader as loader
+
+REQUIREMENTS = ['yamllint>1,<2']
+
+
+def run(args):
+    """Handle ensure config commandline script."""
+    parser = argparse.ArgumentParser(
+        description=("Ensure a Home Assistant config exists, "
+                     "creates one if necessary."))
+    parser.add_argument(
+        '-c', '--config',
+        metavar='path_to_config_dir',
+        default=config_util.get_default_config_dir(),
+        help="Directory that contains the Home Assistant configuration")
+    parser.add_argument(
+        '--lint', default=False,
+        help="Execute Yamllint")
+    parser.add_argument(
+        '--script',
+        choices=['check_config'])
+
+    args = parser.parse_args(args)
+
+    config_dir = os.path.join(os.getcwd(), args.config)
+    # Test if configuration directory exists
+    if not os.path.isdir(config_dir):
+        print('Config directory does not exist:', config_dir)
+        return 1
+
+    yaml_files = ['-c', os.path.splitext(__file__)[0] + '.yaml']
+    yaml_files.extend(glob.glob(os.path.join(config_dir, '*.yaml')))
+    # Python 3.5 gets a recursive, but not in 3.4
+    yaml_files.extend(glob.glob(os.path.join(config_dir, '*/*.yaml')))
+
+    if args.lint:
+        _yaml_ordereddict(False)  # Does not like Ordereddicts
+        import yamllint.cli
+        yamllint.cli.run(yaml_files)
+
+    # Normal loading
+    config_path = config_util.ensure_config_exists(config_dir)
+    print("Normal setup", config_path)
+    _yaml_ordereddict(True)
+
+    component_list = []
+
+    # @patch("homeassistant.bootstrap._setup_component", return_value=True)
+    # @patch("component.setup", return_value=True)
+
+    patchers = []
+
+    def cc_loader_get_component(comp_name):
```

Comment:
Instead of doing this, why not do this:

```python
def mock_get_component(comp_name):
    comp = loader.get_component(comp_name)

    def mock_setup(*args):
        …

    # Test if platform
    if '.' in comp_name:
        comp.setup_platform = mock_setup
    else:
        comp.setup = mock_setup

    return comp
```